### ISSUE TITLE
NFR: Create way to easily activate/deactivate or hide/unhide bar and spinner

### ISSUE THREAD

**Issue by moderndeveloperllc** [vaadin/vaadin-progress-bar#8]
Right now if I want to add an indefinite progress bar that only shows during an action like a search, I need to add and remove the element form the DOM.

To show I'm having to do something like:
```
let progress = document.createElement('vaadin-progress-bar');
progress.id = 'searchProgress';
this.$.searchFormDiv.appendChild(progress);
```
and then when my action finishes, I do
```
this.shadowRoot.querySelector('#searchProgress').remove();
```
not overly intuitive. It would be nice if it had something similar to `paper-spinner` where I can set `hidden` and `active` attributes that govern the visibility and/or active state of the bar. With the ability to have an indefinite `value`, perhaps only something like an `active` property is needed.

**Comment by jouni** [comment]
Hi! I think you could bind to the `hidden` attribute directly, which is available for any element and makes it `display: none;`

`<vaadin-progress-bar hidden$="[[someBooleanProperty]]">`

**Comment by moderndeveloperllc** [comment]
The :host CSS selector prevents that from happening on Chrome. I'll put up a jsbin when I get home.

**Comment by manolo** [comment]
`hidden` is an attribute handled natively by the browser which sets `display: none` as @jouni pointed.
But it has the most low preference in css rules, so any selector targeting the element with any `display` value will always win, and this is the case because de component has the `:host {display: block;}` rule in its template.
The solution is that you need to put a rule in your app element like `vaadin-progress-bar[hidden] {display:none}`

**Comment by manolo** [comment]
By the way, some elements in paper have this css rule
```
      :host([hidden]) {
        display: none !important;
      }
```
It could be something that we could add to all our elements @jouni ?

**Comment by jouni** [comment]
@manolo, yeah I don’t see why we couldn’t.

**Comment by moderndeveloperllc** [comment]
Thanks @manolo, that worked. I look forward to y'all having this across all your elements so I can skip the hack :-)

**Comment by stramel** [comment]
I love referencing this article, [hidden.is.a.lie.fyi](hidden.is.a.lie.fyi) I would definitely suggest adding the 

```css
:host([hidden]) {
  display: none !important;
}
```

**Comment by limonte** [comment]
Fixed in #25
